### PR TITLE
Resolve aliases

### PR DESCRIPTION
**Pull request by HughPowell**
Initial PR for #39 Support :resolve-aliases.
This should work for the extra-deps alias option although I'm not sure about the other two.

### CODE REVIEW

File: src/lein_tools_deps/plugin.clj [hunk 1]
```diff
@@ -79,7 +79,7 @@
   (-> deps-files
       reader/read-deps))
 
-(defmulti leinize (fn [[dep-key dep-val]]
+(defmulti leinize (fn [[_ dep-val]]
```
Comment (RickMoynihan) [inline src/lein_tools_deps/plugin.clj]: @HughPowell I think here it is useful to know the name even though it's ignored... i.e. it makes it clear these are `[key value]` pairs (from `seq`ing a hashmap).

Elsewhere we use the convention `_arg-name`, to provide the hint.  IIRC IDE's respect this as well.

**Comment by RickMoynihan** [comment]
👍 thanks for doing the bulk of this @HughPowell awesome work, and a great new feature!

File: src/lein_tools_deps/plugin.clj (hunk 1 above)
Comment (RickMoynihan) [reply src/lein_tools_deps/plugin.clj]: Can fix this up later though.

Comment (HughPowell) [reply src/lein_tools_deps/plugin.clj]: Yep, didn't know about _arg-name, but looks like a really good idea.

**Pull request merged**